app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt

```diff
@@ -249,22 +263,36 @@ fun ActivityDetailsScreen(
                   modifier = Modifier.padding(bottom = 8.dp))
 
               // List of participants
-              Column {
+              Column(modifier = Modifier.testTag("participants")) {
                 activity?.participants?.forEach { participant ->
                   Row(
                       verticalAlignment = Alignment.CenterVertically,
-                      modifier = Modifier.padding(vertical = 4.dp)) {
-                        // Placeholder for participant picture
-                        Box(
-                            modifier =
-                                Modifier.size(40.dp)
-                                    .background(Color.Gray, shape = RoundedCornerShape(8.dp))
-                                    .padding(8.dp)) {
-                              Text(
-                                  text = "P", // Placeholder for profile picture
-                                  color = Color.White,
-                                  modifier = Modifier.align(Alignment.Center))
+                      modifier =
+                          Modifier.padding(vertical = 4.dp).testTag(participant.name).clickable {
+                            if (participant.name != profile?.name) {
+                              navigationActions.navigateTo(Screen.PROFILE)
                             }
+                          }) {
```

Comment:
We should all be carful in not using magic constants! We shoud change this habit in the next sprint!